client/src/Component/MainTable.js

```diff
@@ -0,0 +1,89 @@
+import React, { useState, useEffect } from "react";
+import moment from "moment";
+import "./MainTableStyle.css";
+
+const MainTable = () => {
+  const [clientData, setClientData] = useState (null);
+  const [error, setError] = useState(null);
+
+  useEffect(() => {
+    fetch("/api/clients")
+      .then((response) => {
+        if (!response.ok) {
+          throw Error("We could not fetch the data for that resource");
+          }
+          return response.json();
+        })
+      .then((data) => {
+        setClientData(data);
+        setError(null);
+      })
+      .catch((err) => {
+              console.log("error message: ", err.message);
+              setError(err.message);
+      });
+  }, []);
+
+if (clientData === null){
+  return (
+    <div className="d-flex justify-content-center">
+      <div className="spinner-grow">
+        <span className="visually-hidden">Loading...</span>
+      </div>
+    </div>
+  );
+}
+  return (
+    <div className="table-responsive{-sm|-md|-lg|-xl|-xxl}">
+      <table className="table table align-middle  table-striped table-hover styled-table border">
+          <caption className="visually-hidden">Clients</caption>
+        <thead className="table-dark">
+          <tr className="">
+            <th scope="col">Id</th>
+            <th scope="col">Added</th>
+            <th scope="col">Name</th>
+            <th scope="col">Phone</th>
+            <th scope="col">Bikes needed</th>
+            <th scope="col">Booking status</th>
+            <th scope="col">Residency status</th>
+            <th scope="col">Country of origin</th>
+            <th scope="col">Time in Scotland</th>
+            <th scope="col">Language spoken</th>
+            <th scope="col">English speaker</th>
+            <th scope="col">English skill level</th>
+            <th scope="col">Gender</th>
+            <th scope="col">DOB</th>
+            <th scope="col">Postcode</th>
+            <th scope="col">Referring agency</th>
+          </tr>
+        </thead>
+        <tbody className="table-striped">
+          {clientData.map((client) => {
+            return (
+              <tr key={client.client_id}>
+                <td>{client.client_id}</td>
+                <td>{moment(client.date_added).format("L")}</td>
```

Comment:
```suggestion
                <td>{new Date(client.date_added).toLocaleString()}</td>
```
We don't need to use a library to format the date—let's format the date using plain JavaScript.

Libraries add extra size to our application which means it takes longer for users to load the application, so we should be careful when choosing libraries.

Reply:
I got it Coung, less weight fast load.